injection/sharedmain/main.go

```diff
@@ -106,17 +106,13 @@ func GetLeaderElectionConfig(ctx context.Context) (*leaderelection.Config, error
 
 // GetObservabilityConfig gets the observability config from the (in order):
 // 1. provided context,
-// 2. reading from the API server,
-// 3. defaults (if not found).
+// 2. reading from the API server
+// It errors if the ObservabilityConfig is NotFound
 func GetObservabilityConfig(ctx context.Context) (*metrics.ObservabilityConfig, error) {
 	if cfg := metrics.GetObservabilityConfig(ctx); cfg != nil {
 		return cfg, nil
 	}
-
 	observabilityConfigMap, err := kubeclient.Get(ctx).CoreV1().ConfigMaps(system.Namespace()).Get(ctx, metrics.ConfigMapName(), metav1.GetOptions{})
-	if apierrors.IsNotFound(err) {
-		return metrics.NewObservabilityConfigFromConfigMap(nil)
-	}
```

Comment:
Do we really want to do that? That requires everyone using this package to now introduce the `METRICS_DOMAIN` environment variable into their containers with no option to opt-out, which is obviously breaking for anyone who takes dependency on `sharedmain` initialization?

Reply:
If I recall correctly `METRICS_DOMAIN` was used in a stack driver exporter that's now removed - so I think we can remove it.

Reply:
has stack driver support been removed? 

I still see it here, https://github.com/knative/serving/blob/main/pkg/queue/sharedmain/main.go#L550

can we provide a default value for `METRICS_DOMAIN` in this context? 
or even a placeholder value if the backend is for prometheus, which I don't think uses this value.

Reply:
For the OpenCensus metrics exporter, I'm not sure I can think of a good default value beyond something with knative in it, but that might not be very observable and is very easy to get the configuration wrong.

Granted, a large number of people are probably using Prometheus over other exporters, so I agree that a default value containing something like `knative/pkg` may be the most reasonable thing here to prevent breaking users

Reply:
> 

thats my vote, to default to something like `knative/pkg`, feels weird to fail on an env value even if it is not being used with the prometheus configuration.

or would it possible to only panic on `Domain()` if `opencensus` is specified?

Reply:
Updated after your change @dprotaso #2617